python/tank/platform/constants.py

```diff
@@ -175,6 +175,12 @@
 # the name of the shell engine
 SHELL_ENGINE = "tk-shell"
 
+# valid characters for a template key name
+TEMPLATE_KEY_NAME_REGEX = "[a-zA-Z_ 0-9\.]+"
```

Comment:
I can't say I am a fan of catch-all files that hold unrelated pieces. Unless this needs to be shared across multiple files, I am not sure I like the fact that this becomes suddenly global when it's a local feature.

It seems like it is being use solely by the templating system, therefore it should remain in that general area.

What I am concerned about is this pattern is that it becomes difficult to clean up something when you are removing it or updating it. What could have been a very centralized fix now becomes a hunt for all the dangling pieces.